src/NuGet.Core/NuGet.PackageManagement/IDE/IPackageRestoreManager.cs

```diff
@@ -78,6 +86,21 @@ public interface IPackageRestoreManager
             INuGetProjectContext nuGetProjectContext,
             CancellationToken token);
 
+        /// <summary>
+        /// Restores the solution if the assets file is missing.
+        /// </summary>
+        /// <param name="solutionDirectory"></param>
+        /// <param name="nuGetProjectContext"></param>
+        /// <param name="logger"></param>
+        /// <param name="token"></param>
+        /// <returns>Returns true if the restore succeded</returns>
+        Task<PackageRestoreResult> RestoreMissingAssetsFileInSolutionAsync(string solutionDirectory,
```

Comment:
Our VS code already has code/methods to restore the solution. It's in NuGet.SolutionRestoreManager, and it makes sure that we don't have concurrent restores running. My gut feeling is that we should not need to add new restore APIs.